Aplib.Core/Belief/ListBelief.cs

```diff
@@ -23,49 +23,62 @@ public class ListBelief<TReference, TObservation> : Belief<IEnumerable<TReferenc
     {
         /// <summary>
         /// Initializes a new instance of the <see cref="ListBelief{TReference,TObservation}" /> class from an object
-        /// reference collection, and a function to generate an observation from an object reference.
+        /// reference collection, a function to generate an observation from an object reference, and optionally an
+        /// update guard.
         /// </summary>
+        /// <param name="metadata">
+        /// Metadata about this Belief, used to quickly display the goal in several contexts.
+        /// </param>
         /// <param name="references">
         /// The collection of reference objects. The underlying type implementing <see cref="IEnumerable{TReference}" />
         /// <i>must</i> be a reference type, note that this is not enforced by C#.
         /// </param>
         /// <param name="getObservationFromReference">
         /// A function that takes an object reference and generates an observation.
         /// </param>
+        /// <param name="shouldUpdate">A condition on when the observation should be updated.</param>
         /// <exception cref="ArgumentException">
         /// Thrown when <paramref name="references"/> is not a reference type.
         /// </exception>
         public ListBelief
         (
+            Metadata metadata,
             IEnumerable<TReference> references,
-            Func<TReference, TObservation> getObservationFromReference
+            Func<TReference, TObservation> getObservationFromReference,
+            Func<bool> shouldUpdate
         )
-            : base(references, refer => refer.Select(getObservationFromReference).ToList())
+            : base(metadata, references, refer => refer.Select(getObservationFromReference).ToList(), shouldUpdate)
         {
         }
 
-        /// <summary>
-        /// Initializes a new instance of the <see cref="ListBelief{TReference,TObservation}" /> class from an object
-        /// reference collection, a function to generate an observation from an object reference, and an update guard.
-        /// </summary>
-        /// <param name="references">
-        /// The collection of reference objects. The underlying type implementing <see cref="IEnumerable{TReference}" />
-        /// <i>must</i> be a reference type, note that this is not enforced by C#.
-        /// </param>
-        /// <param name="getObservationFromReference">
-        /// A function that takes an object reference and generates an observation.
-        /// </param>
-        /// <param name="shouldUpdate">A condition on when the observation should be updated.</param>
-        /// <exception cref="ArgumentException">
-        /// Thrown when <paramref name="references"/> is not a reference type.
-        /// </exception>
+        /// <inheritdoc cref="ListBelief{TReference,TObservation}(Metadata,IEnumerable{TReference},Func{TReference,TObservation},Func{bool})"/>
```

Comment:
Again this mentions MetaData, and thus inherits and displays documentation for a param which is not present. I will not mention it any more, but I request you to remove these redundant xml docs everywhere.

Reply:
This is a Rider bug. It renders correctly in docfx and in Visual Studio. I won't do a workaround for this Rider bug because the only workaround would involve an enormous amount of documentation duplication.